pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -384,10 +384,20 @@ public int size() {
         return payload.readableBytes();
     }
 
-    public Schema<T> getSchema() {
+    public Schema<T> getSchemaInternal() {
         return this.schema;
     }
 
+    @Override
+    public Optional<Schema<?>> getSchema() {
+        ensureSchemaIsLoaded();
+        if (schema == null) {
+            return Optional.empty();
+        }
+        byte[] schemaVersion = getSchemaVersion();
+        return Optional.of(schema.atSchemaVersion(schemaVersion));
```

Comment:
depending on what happens with unknown schema version, this can throw exception or may need .ofNullable

Reply:
here we are consuming a Message, so if we have a version this version must exist

I believe it is good to see a RuntimeException in this impossible case